Tests/FluentAssertions.Specs/Primitives/BooleanAssertionSpecs.cs

```diff
@@ -6,151 +6,200 @@ namespace FluentAssertions.Specs.Primitives;
-            .WithMessage("Expected boolean to be false because we want to test the failure message, but found True.");
-    }
-
-    [Fact]
-    public void Should_succeed_when_asserting_boolean_value_to_be_equal_to_the_same_value()
-    {
-        // Act
-        Action action = () =>
-            false.Should().Be(false);
-
-        // Assert
-        action.Should().NotThrow();
-    }
-
-    [Fact]
-    public void Should_fail_when_asserting_boolean_value_to_be_equal_to_a_different_value()
-    {
-        // Act
-        Action action = () =>
-            false.Should().Be(true);
-
-        // Assert
-        action.Should().Throw<XunitException>();
-    }
-
-    [Fact]
-    public void Should_fail_with_descriptive_message_when_asserting_boolean_value_to_be_equal_to_a_different_value()
-    {
-        // Act
-        Action action = () =>
-            false.Should().Be(true, "because we want to test the failure {0}", "message");
-
-        // Assert
-        action.Should().Throw<XunitException>()
-            .WithMessage("*Expected*boolean*True*because we want to test the failure message, but found False.*");
-    }
-
-    [Fact]
-    public void Should_succeed_when_asserting_boolean_value_not_to_be_equal_to_the_same_value()
-    {
-        // Act
-        Action action = () =>
-            true.Should().NotBe(false);
-
-        // Assert
-        action.Should().NotThrow();
-    }
-
-    [Fact]
-    public void Should_fail_when_asserting_boolean_value_not_to_be_equal_to_a_different_value()
-    {
-        // Act
-        Action action = () =>
-            true.Should().NotBe(true);
-
-        // Assert
-        action.Should().Throw<XunitException>();
-    }
-
-    [Fact]
-    public void Should_fail_with_descriptive_message_when_asserting_boolean_value_not_to_be_equal_to_a_different_value()
-    {
-        // Act
-        Action action = () =>
-            true.Should().NotBe(true, "because we want to test the failure {0}", "message");
-
-        // Assert
-        action.Should().Throw<XunitException>()
-            .WithMessage("*Expected*boolean*True*because we want to test the failure message, but found True.*");
-    }
-
-    [Fact]
-    public void Should_throw_a_helpful_error_when_accidentally_using_equals()
-    {
-        // Act
-        Action action = () => true.Should().Equals(true);
-
-        // Assert
-        action.Should().Throw<NotSupportedException>().WithMessage("Equals is not part of Fluent Assertions. Did you mean Be() instead?");
+        public static TheoryData<bool?, bool> PassingImplications => new()
+        {
+            { false, false },
+            { false, true },
+            { true, true }
+        };
+
+        public static TheoryData<bool?, bool> NonPassingImplications => new()
+        {
+            { null, true },
+            { null, false },
+            { true, false }
+        };
+
+        [Theory]
+        [MemberData(nameof(PassingImplications))]
```

Comment:
🤔 I think using `[InlineData]` is much clearer in this particular case and keeps the data with the test. Is there a reason why we can't use those?

Reply:
There is no reason to not use `[InlineData]`, I just like the feature of `TheoryData`.. 

And I think it is (especially for new contributors) easier, if there is one "design" how to structure tests. 

But if @jnyrup agrees to the other approach I am willing to change that..

Reply:
> And I think it is (especially for new contributors) easier, if there is one "design" how to structure tests.

I'm in the "use the right tool for the right job at the right time" camp ;-).

Reply:
I had _completely_ forgotten about the simpler `InlineData` when reviewing this 😄

My thoughts on when to pick which:
* Aim for `[InlineData]` as it keeps the parameters closest to the method body.
* If the parameters are not compile-constants, defer to `TheoryData` defined right below the method using it.
* If the parameters have a _good_ reason to be shared between multiple tests, `TheoryData` might do the best job. There's a thin line here between DRY and over-DRY and we tend to prefer duplication over coupling between tests, that's why you'll see very few "helper" methods in the test suite.

Reply:
Ok... thanks for clarification :)